IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetrySerializedItem.m

```diff
@@ -0,0 +1,110 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDCurrentRequestTelemetrySerializedItem+Internal.h"
+#import "MSIDLastRequestTelemetrySerializedItem.h"
+
+@interface MSIDLastRequestTelemetrySerializedItem()
+
+@property (nonatomic) NSArray<MSIDRequestTelemetryErrorInfo *> *errorsInfo;
+
+@end
+
+@implementation MSIDLastRequestTelemetrySerializedItem
+
+- (instancetype)initWithSchemaVersion:(NSNumber *)schemaVersion defaultFields:(NSArray *)defaultFields errorInfo:(NSArray *)errorsInfo platformFields:(NSArray *)platformFields
+{
+    self = [super initWithSchemaVersion:schemaVersion defaultFields:defaultFields platformFields:platformFields];
+    if (self)
+    {
+        _errorsInfo = errorsInfo;
+    }
+    return self;
+}
+
+// Builds last telemetry string using default serialization of each set of fields
+// specified in the last telemetry string schema
+- (NSString *)serialize
+{
+    NSString *telemetryString = [NSString stringWithFormat:@"%@|", self.schemaVersion];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [super serializeFields: super.defaultFields]];
+    
+    NSUInteger startLength = [telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [self serializeErrorsInfoWithCurrentStringSize:startLength]];
+    
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@", [super serializeFields: self.platformFields]];
+    
+    return telemetryString;
+}
+
+#pragma mark Helper
+
+- (NSString *)serializeErrorsInfoWithCurrentStringSize:(NSUInteger)startLength
+{
+    NSString *failedRequestsString = @"";
+    NSString *errorMessagesString = @"";
+    
+    if (self.errorsInfo.count > 0)
+    {
+        int lastIndex = (int)self.errorsInfo.count - 1;
+        
+        // Set first post in fencepost structure -- last item in string doesn't have comma at the end
+        NSString *currentFailedRequest = [NSString stringWithFormat:@"%ld,%@", self.errorsInfo[lastIndex].apiId, self.errorsInfo[lastIndex].correlationId ?: @""];
```

Comment:
This line triggers a warning:

```
Values of type 'NSInteger' should not be used as format arguments; add an explicit cast to 'long' instead
Replace '%ld,%@", ' with '%ld,%@", (long)'
```

Would be great if you could fix it. Thanks.

Reply:
Fixed, thanks for pointing it out. Is there a setting that I have misconfigured that I didn't see this warning? Just so I can avoid pushing code with these kinds of warnings in the future.

Reply:
I think we're just missing a setting that is present in the Authenticator project. Worth following up but doesn't need to be in this PR. Thanks.